iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/GeneratorXml.java

```diff
@@ -92,7 +92,11 @@ protected void readSubElements(Generator g, NetworkXmlReaderContext context) thr
                 case "regulatingTerminal":
                     String id = context.getAnonymizer().deanonymizeString(context.getReader().getAttributeValue(null, "id"));
                     String side = context.getReader().getAttributeValue(null, "side");
-                    context.getEndTasks().add(() -> g.setRegulatingTerminal(TerminalRefXml.readTerminalRef(g.getTerminal().getVoltageLevel().getSubstation().getNetwork(), id, side)));
+                    if (id != null) {
+                        context.getEndTasks().add(() -> g.setRegulatingTerminal(TerminalRefXml.readTerminalRef(g.getTerminal().getVoltageLevel().getSubstation().getNetwork(), id, side)));
+                    } else {
+                        context.getEndTasks().add(() -> g.setRegulatingTerminal(TerminalRefXml.readTerminalRef(g.getTerminal().getVoltageLevel().getSubstation().getNetwork(), g.getId(), side)));
+                    }
```

Comment:
I think we can write this code differently: why don't we pass a default value to the `getAttributeValue(null, "id", g.getId())` and remove this `if/else`?

I propose to make this change everywhere

Reply:
Because of the deanonymize we will have an error if we define a new getAttributeValue.

Is it convenient like this:
String id = StringUtils.defaultIfEmpty(context.getAnonymizer().deanonymizeString(context.getReader().getAttributeValue(null, "id")), g.getId());

Reply:
```
String id = Optional.ofNullable(context.getAnonymizer().deanonymizeString(context.getReader().getAttributeValue(null, "id"))).orElse(g.id());
```